nemo/collections/nlp/models/language_modeling/lm_model.py

```diff
@@ -162,64 +159,65 @@ def validation_epoch_end(self, outputs):
         Called at the end of validation to aggregate outputs.
         :param outputs: list of individual outputs of each validation step.
         """
-        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
-        return {'val_loss': avg_loss}
+        if outputs:
```

Comment:
why do we need this?

Reply:
When I run without validation_ds, i get an error:
```
Validation sanity check: 0it [00:00, ?it/s]Traceback (most recent call last):
  File "bert_pretraining_from_text.py", line 31, in main
    trainer.fit(bert_model)
  File "/opt/conda/lib/python3.6/site-packages/pytorch_lightning/trainer/trainer.py", line 992, in fit
    results = self.spawn_ddp_children(model)
  File "/opt/conda/lib/python3.6/site-packages/pytorch_lightning/trainer/distrib_data_parallel.py", line 462, in spawn_ddp_children
    results = self.ddp_train(local_rank, q=None, model=model, is_master=True)
  File "/opt/conda/lib/python3.6/site-packages/pytorch_lightning/trainer/distrib_data_parallel.py", line 560, in ddp_train
    results = self.run_pretrain_routine(model)
  File "/opt/conda/lib/python3.6/site-packages/pytorch_lightning/trainer/trainer.py", line 1196, in run_pretrain_routine
    False)
  File "/opt/conda/lib/python3.6/site-packages/pytorch_lightning/trainer/evaluation_loop.py", line 342, in _evaluate
    eval_results = model.validation_epoch_end(outputs)
  File "/home/eharper/github/new_NeMo/NeMo/nemo/collections/nlp/models/language_modeling/lm_model.py", line 163, in validation_epoch_end
    avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
RuntimeError: stack expects a non-empty TensorList
```